resources/qml/Menus/MaterialBrandMenu.qml

```diff
@@ -243,11 +250,37 @@ Cura.MenuItem
                         height: materialColorsList.height + padding * 2
                         x: parent.width
                         y: {
-                            // If flipped the popup should push up rather than down from the parent
-                            if (brandMaterialBase.isFlipped) {
-                                return -height + UM.Theme.getSize("menu").height + UM.Theme.getSize("default_lining").width
+                            // This popup will always try to stay within the vertical space of the parent of MaterialBrandMenu
+                            // If it is larger than the parent, it will expand downwards.
+                            var popupHeight = model.colors.count * UM.Theme.getSize("menu").height
+                            var spaceToBottom = materialTypesList.height - parent.y // Space from hovered item to bottom of list
+
+                            if (popupHeight < spaceToBottom)
+                            {
+                                return -UM.Theme.getSize("default_lining").width
+                            }
+                            else
+                            {
+                                var yAlignedWithTopOfRootPopup = - materialBrandMenu.y - UM.Theme.getSize("default_lining").height
+
+                                if (popupHeight > rootHeight && popupHeight > menuPopup.height)
```

Comment:
I strongly believe in capturing values as "documenting variables". The comment below explains very well what is happening, but also having the variable name telling exactly what is happening helps in my opinion.
```suggestion
                                const parentHeight = menuPopup.height;
                                if (popupHeight > rootHeight && popupHeight > parentHeight)
```